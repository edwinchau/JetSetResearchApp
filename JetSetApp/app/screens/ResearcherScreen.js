 import React, {Component, Fragment} from 'react';
import { Text, View, TextInput, Button, ScrollView, TouchableOpacity, TouchableElement} from 'react-native';

import styles from '../resources/Styles'
import SaveData from '../resources/SaveData';
import SendData from '../resources/SendData';

export default class ResearcherScreen extends Component {

    static navigationOptions = () => {
        return {
            headerStyle: styles.headerStyle,
            headerTintColor: 'black',
            headerTitle: 'Researcher Menu',
            headerTitleStyle: {
                flex: 1,
            }
        };
    };


    state = {
        typedPassword: '',
        adminPassword: 'abc123'
    };

    handlePassword = (text) => {
        this.setState({typedPassword: text});
    }

    checkPassword = () => {
        if (this.state.adminPassword !== this.state.typedPassword) {
            return false;
        }
        return true;
    }

    deleteAllFiles = () => {
        if (this.checkPassword()) {
            SaveData.deleteAllFiles();
            alert("Deleted All Files");
        } else {
            alert("Incorrect password");
        }
    }

    sendEmail = () => {
        if (this.checkPassword()) {
            SendData.sendDataEmail();
        } else {
            alert("Incorrect password");
        }
    }

    render() {
        return (
            <View style={styles.backgroundAdmin}>
                <ScrollView contentContainerStyle={{flexGrow: 1}} keyboardShouldPersistTaps='handled'>

                    <View style = {styles.elementPadding}>
                        <TextInput
                            placeholder=" Please Type in Admin Password First "
                            style = {styles.passwordTextbox}
                            secureTextEntry={true}
                            onChangeText={this.handlePassword}
                        />
                    </View>

                    <View style = {styles.elementPadding}>
                        <TouchableOpacity
                            style={styles.adminButton}
                            onPress={() => this.deleteAllFiles()}
                        >
                            <Text style={styles.buttonTextColour}> Delete All Files </Text>
                        </TouchableOpacity>
                    </View>

                    <View style = {styles.elementPadding}>
                        <TouchableOpacity
                        style={styles.adminButton}
                        onPress={() => this.sendEmail()}>
                            <Text style={styles.buttonTextColour}> Send Email </Text>
                        </TouchableOpacity>
                    </View>
                </ScrollView>

            </View>
        )
    }
}
