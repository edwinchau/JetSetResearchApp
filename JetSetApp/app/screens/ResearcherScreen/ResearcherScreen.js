import React, {Component, Fragment} from 'react';
import { StyleSheet, Text, View, TouchableOpacity, TextInput} from 'react-native';

import styles from './Styles'
import SaveData from '../../resources/SaveData/SaveData';
import SendData from '../../resources/SendData/SendData';

export default class ResearcherScreen extends Component {


    state = {
        typedPassword: '',
        adminPassword: 'abc123'
    };

    handlePassword = (text) => {
        this.setState({typedPassword: text});
    }

    checkPassword = () => {
        if (this.state.adminPassword !== this.state.typedPassword) {
            return false; 
        }
        return true;
    }

    deleteAllFiles = () => {
        if (this.checkPassword()) {
            SaveData.deleteAllFiles();
            alert("Deleted All Files");
        } else {
            alert("Incorrect password");
        }
    }

    sendEmail = () => {
        if (this.checkPassword()) {
            SendData.sendDataEmail();
        } else {
            alert("Incorrect password");
        }
    }

    render() {
        return (
            <View style={styles.container}>
                <TextInput
                    placeholder="Please Type in Admin Password First"
                    secureTextEntry={true}
                    onChangeText={this.handlePassword}
                />
                <TouchableOpacity style={styles.button}
                    onPress={() => this.deleteAllFiles()
                    }>
                    <Text style={styles.submitButtonText}> Delete All Files </Text>
                </TouchableOpacity>   
                <TouchableOpacity style={styles.button}
                    onPress={() => this.sendEmail()
                    }>
                    <Text style={styles.submitButtonText}> Send Data </Text>
                </TouchableOpacity>      
                <TouchableOpacity style={styles.button}
                    onPress={() => this.prompt()
                    }>
                    <Text style={styles.submitButtonText}> Prompt </Text>
                </TouchableOpacity>         
            </View>
        )
    }
}
