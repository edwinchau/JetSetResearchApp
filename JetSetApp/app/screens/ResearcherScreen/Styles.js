import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'purple'
    },
    input: {
        height: 40, borderColor: 'gray', borderWidth: 1
    },
    button: {
        backgroundColor: 'green',
        borderColor: 'green',
        borderWidth: 1,
        fontSize: 24,
        fontWeight: 'bold',
        overflow: 'hidden',
        padding: 12,
        textAlign:'center',        
    }
});

module.exports = styles;
