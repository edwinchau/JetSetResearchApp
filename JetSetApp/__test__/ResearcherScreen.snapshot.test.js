import React from "react";
import ResearcherScreen from "../app/screens/ResearcherScreen"
import renderer from "react-test-renderer"

describe("<ResearcherScreen/>", ()=>{
  it("snapshot", async ()=>{
    expect(renderer.create(<ResearcherScreen/>).toJSON()).toMatchSnapshot();
  });
});
