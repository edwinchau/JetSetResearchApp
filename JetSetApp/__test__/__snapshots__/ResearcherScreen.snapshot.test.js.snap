// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResearcherScreen/> snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#DF4425",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    keyboardShouldPersistTaps="handled"
  >
    <View>
      <View
        style={
          Object {
            "paddingVertical": 5,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder=" Please Type in Admin Password First "
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "black",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginLeft": 10,
              "marginRight": 10,
              "maxWidth": "90%",
              "minWidth": "70%",
              "padding": 10,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "paddingVertical": 5,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "black",
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          >
             Delete All Files 
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "paddingVertical": 5,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "black",
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          >
             Send Email 
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
